agent/cache-types/connect_ca_leaf.go

```diff
@@ -469,7 +469,7 @@ func activeRootHasKey(roots *structs.IndexedCARoots, currentSigningKeyID string)
 }
 
 func (c *ConnectCALeaf) rootsFromCache() (*structs.IndexedCARoots, error) {
-	rawRoots, _, err := c.Cache.Get(ConnectCARootName, &structs.DCSpecificRequest{
+	rawRoots, _, err := c.Cache.Get(context.Background(), ConnectCARootName, &structs.DCSpecificRequest{
```

Comment:
It might be possible to get a context here, but if we want to delay it, `context.TODO()` will make it easier to find these places where we need to do more plumbing.

Reply:
Here, the index is specifically not set so the request wont block in the same way. Thats why I used background as it will not block.

Reply:
Going to add a comment about this.